\normalsize
%\textcolor{red}{I think I shold discuss more the different biomarker time-telling approaches that have been imlemented and in which tissues so far -- in the intro I don't have time/space for this: Dijk blood biomarkers, Ueda pioneer... Also sex not contributing}\\

%%Discuss figure 1
We report here a study of circadian gene expression in the skin of 11 \textit{healthy} human subjects sampled longitudinally every 4\,h over a 24\,h period. Unlike previous genome-wide analyses of the human skin \cite{Wu2018,Wu2020}, this dataset contains samples at higher temporal resolution with chronobiological profiles of all study subjects. This latter information allowed us to assess the chronotypes of subjects (as mid-sleep time on free days corrected for sleep debt) and thereby perform all analyses with respect to internal time. The diversity of chronotypes or phases of entrainment in the human population \cite{Roenneberg2007} necessitates such an approach. In other words, molecular rhythms across subjects are expected to be coherent only after chronotype differences are corrected for. Nevertheless, most studies continue to assess population rhythms with respect to wall time (or sample collection times) \cite{Laing2019}. We attempted to quantify the difference in our results between the use of internal and wall times, but the 4\,h sampling resolution is still insufficient to accurately reflect the $\sim$6h range of 95\% of human chronotypes.

The cellular complexity and heterogeneity of human skin suggested that the notion of a singular ``skin clock'' is too simplistic. We therefore determined circadian gene expression in two important skin layers, the dermis and epidermis, in all subjects. We presented a hierarchical description of circadian rhythms in the two layers. First, the population gene expression rhythms represent average rhythms across the cohort. Conceptually, the population rhythms are an estimate of the average rhythms of a hypothetical population of individuals, from which the cohort is randomly sampled. Second, we quantified the variability of circadian rhythms across the cohort as a representation of variability within the hypothetical population.

Our analysis revealed $\sim$1400 circadian genes in either layer at the population level, significantly expanding on the list of known clock output genes in human skin \cite{Wu2018, Wu2020}. However, two-thirds of these genes had indistinguishable rhythms between the layers. On the one hand, this is unsurprising given the physical proximity between the layers. On the other, it is unexpected given the well-documented skin heterogeneity and tissue-specificity of circadian programs in physiology \cite{Ruben2018}. As reported in many studies, the core clock gene expression was consistent between the layers with the exception of \textit{ARNTL} and \textit{PER3}. Despite this general similarity, a third of the circadian genes did display expression in only one or the other layer. On the whole, the epidermal circadian genes tended to have higher amplitudes than the dermal circadian genes (as suggested previously \cite{Wu2020}), which is also reflected in greater number of circadian genes in the epidermis compared to the dermis. We can only speculate that this might be because of the direct environmental exposure of the epidermis to external Zeitgebers.


%Here we report a novel human dataset in which we examined circadian gene expression of skin in 11 subjects that were sampled longitudinally every 4\,h for 24\,h. Although skin circadian rhythms have been analyzed in previous studies in a genome-wide manner \cite{Wu2018, Wu2020}, this dataset has a higher sampling frequency and contains information about the subjects, what allows to control for external factors and to assess what the sources of variation in such a human study might be. We identified the circadian transcriptome of human dermis and epidermis with respect to \textit{internal} time by correcting external time to each individual's mid sleep time. We found that roughly a fifth of the circadian skin transcriptome ($\sim$280 out of 1400 rhythmic genes) is shared between both layers, while a large part is layer-specific. Interestingly, and despite their physical proximity, we found amplitude and phase differences in the core clock and clock-controlled genes in both layers. In general, epidermis is earlier in phase and displays higher amplitude rhythms (Figures \ref{fig:fig1}D-F), with some exceptions like \textit{ARNTL, DBP} or \textit{PER1} that oscillate with higher amplitudes in dermis.\\ %\textcolor{red}{Such results might be translatable to larger population studies as well}. 

%Circadian timing mechanisms are sensitive to day length and temperature, and therefore the clock-controlled transcriptome represents a candidate for the regulation of seasonal phenomena within the skin. Interestingly, several skin diseases have been shown to exhibit seasonal change in severity \cite{Weiss2008}. Nevertheless, given the cellular complexity and heterogeneity of human skin, it is probably simplistic to talk about a singular ``skin clock''. Our results show that epidermis presents, overall, larger amplitude rhythms compared to dermis. We can only speculate that this might be because of the direct environmental exposure of the epidermis to external Zeitgebers. Could it be that epidermal clocks entrain more efficiently and display higher amplitude resonance, resulting in the observed higher amplitude rhythms? Do amplitude rhythms in skin change with seasons? Skin cancer progression has been shown to be under clock control \cite{Gaddameedhi2011} and possibly affected by feeding schedules, as suggested by a study from 2017, where the authors proposed that time restricted feeding influences sensitivity to UV-induced DNA damage \cite{Wang2017}. Moreover, the amount of caloric intake affects gene expression and function of the skin \cite{Forni2017}. Could it be that skin clocks contain potential information about circadian rhythmicity in additional organs, or vice versa? All these remain important and open questions. \\ %kin is one of the organs directly exposed to external cues (temperature changes, light\slash dark cycles, etc.), || 
%Because the human skin transcriptome is in part under circadian control and skin is one of the organs directly exposed to external cues (temperature changes, light\slash dark cycles, etc.), it might potentially contain information about the phase of circadian rhythmicity in additional organs. Nevertheless, given the cellular complexity and heterogeneity of human skin, it is probably simplistic to talk about a singular ``skin clock''. Our results show that epidermis presents, overall, larger amplitude rhythms compared to dermis. We can only speculate that this might be because of the direct environmental exposure of the epidermis to external Zeitgebers. Could it be that epidermal clocks entrain more efficiently and display higher amplitude resonance, resulting in the observed higher amplitude rhythms? These remain important and open questions. \\

 %\textcolor{red}{What about the Janich-Benitah stories? Also diseases: psoriasis, skin cancer -- Gaddameehdi2011, ageing... (reviewed in \href{https://pubmed.ncbi.nlm.nih.gov/34535902/}{\textbf{Duan-Andersen2021}) -- maybe talk about diseases in discussion?} Circadian timing mechanisms are also sensitive to day length and temperature, and therefore circadian clock mechanisms are candidates for the regulation of seasonal phenomena within the skin. Interestingly, several skin diseases do exhibit seasonal change in severity \href{https://pubmed.ncbi.nlm.nih.gov/18755376/}{Weiss et al 2008}}\\

%%Discuss figure 2
A fundamental challenge in the analysis of high-throughput datasets is to quantify and interpret the contribution of different sources of variation. How does the population or how do the different skin layers affect the genetic regulation of rhythmic gene expression? What are the major drivers of variability? Are there rhythm-specific differences across subjects or skin layers? This set, in which meta-data was available (sex, mid sleep time, age, etc.), allowed to study these kind of questions. We used \texttt{variancePartition}, a publicly available software that leverages the power of linear mixed models, to partition and quantify the contribution of each meta-data variable in the experimental design, plus a residual variance. In our case, we took the rhythmic skin transcripts and analyzed variability in mean expression \textit{across} time, \textit{across} individuals and \textit{across} skin layers as well as the contribution of variance \textit{within }layers and individuals. We defined \textit{across}-layer (or -subject or -time) variability as the variance in mean expression (i.e., magnitude) between dermis and epidermis (inter-layer mean variation in Figure \ref{fig:fig2}). \textit{Within}-layer (or -subject) variability was defined as the variability in rhythms between layers, i.e., the inter-layer circadian variation.\\

We found that differences in magnitude across layers (inter-layer mean variation in Figure \ref{fig:fig2}) represent the strongest source of variance in our cohort, followed by differences in mean expression across subjects and across time. Moreover, when analyzing previously published circadian skin transcriptomic datasets (GSE139300 \cite{GSE139300} and GSE112660 \cite{GSE112660}) we observed that the design variables that contribute most to magnitude variability are also differences between layers, followed by subjects, with large residual variance (not shown \textcolor{red}{maybe remove from results?}). Taken together, these insights suggest (i) that the magnitude of clock-controlled genes varies largely between skin layers and subjects, a novel observation which has, to our knowledge, not been reported previously; and (ii) that such observations are also the case in other population studies and thus might be translatable to larger human cohorts. \\

Although not included in these results, we also assessed whether MST (proxy for chronotype) and sex contributed to magnitude variations in our data, but found almost no contribution. In principle, this speaks for population sampling from skin providing a good estimation of circadian time, regardless of wall time or internal time. Nevertheless, the fact that we observed almost no variance due to chronotype differences might have to do with the fact that the range of MST in our cohort ($\sim$4\,h, Figure \ref{fig:fig1}B, consistent with previous studies \cite{Roenneberg2007}) is in the order of our sampling time. Thus, this experimental design might not be enough to capture subtle chronotype differences and could be a reason explaining why the analyses of internal and external time yield similar results (Suppplementary Figure \ref{fig:suppfig1}). To discern whether the similarity in the internal versus external time analyses of the skin circadian transcriptome is a general property in any cohort or something particular from this dataset, an ideal population study should be sampled at a higher frequency but this poses a complication for obvious practical reasons. \\%we hope to have captured a normal-enough range of chronotypes. 

%%Discuss figure 3
Over the last years, a number of novel approaches have been introduced to assess circadian parameters and, in particular, circadian phase in humans (see \cite{Dijk2020} for a nice review). Some of these methodologies use machine-learning approaches to extract features that predict circadian parameters from high-dimensional -omics data, whereas others are based on collection of data from wearable devices. In this work, we propose a viable set of biomarkers that can accurately predict molecular clock phase in human skin. But, what makes a biomarker a good biomarker? First, a biomarker should be stable and show little variation in magnitude (mean levels) across subjects. Second, it should show a strong temporal variation (that might be specific to the tissue of interest, but not necessarily). It should have relatively strong amplitude rhythms as well as the correct phase. We have shown that a few transcriptome samples contain enough information to predict the phase of \textit{internal} skin rhythms with a small set of candidate genes, whose mean levels are roughly invariant across subjects while showing strong temporal variation (Supplementary Figure \ref{fig:suppfig5}B). We have shown that internal time is predicted more accurately if it is done in both layers separately, as seen by the lower prediction error in Figure \ref{fig:fig3}A compared to Supplementary Figure \ref{fig:suppfig4}A. \textcolor{red}{We have achieved an accuracy (MAE$\sim$0.04\,h) similar to that of the current gold-standard, DLMO (0.5-1\,h \cite{Klerman2002, Danilenko2014})}. Nevertheless, the fact that circadian phase is predicted more accurately in human dermis and epidermis separately unfortunately represents an experimental limitation, since it means that layers must be separated prior to analysis. It should be noted, however, that despite the better performance when done separately, none of the time-telling genes were dermis- or epidermis-specific and just one, \textit{OVGP}, was among the top genes with layer-specific rhythms. \\% We have shown that internal time can be better predicted if epidermis and dermis are previously separated (as seen by the lower MAE in the time prediction when ZeitZeiger is run in each layer separately and not in the skin as a whole), although this poses a complication for practical reasons.\\

We have argued that predicting \textit{internal} time is more appropriate if any evaluation of circadian phase of biomarkers is desired; nevertheless, there are little differences between the rhythmic genes (and their phase and amplitude) obtained when analyzing rhythmicity with respect to internal or external time (Suppplementary Figure \ref{fig:suppfig1}). Thus, we in principle expect the same biomarker set to properly predict internal time in any cohort of individuals even if their chronotype information remains unknown. In this regard, firstly, we observed that the set of biomarkers and their coefficients did not change significantly when ZeitZeiger was run with external instead of internal time (data not shown). Secondly, among our time-telling genes, we found some that are in agreement with previous transcriptomic studies done in human skin with no information about chronotype (\textit{ZBTB16, FKBP5, TRIM35, PER3} and \textit{ARNTL}) \cite{Wu2018, Wu2020}. These studies used a hybrid experimental design in which they combined data from human subjects that were sampled throughout the 24\,h cycle and data from larger population that were sampled just once. The way in which the authors ordered in time and assigned a phase to the samples that were taken just once takes into account internal time. Nevertheless, no chronotype information was available for the longitudinal data, and thus whether the time-telling biomarkers predict internal or wall time was not clear. The authors also validated their biomarker set against the clock time at which the sample was taken, which is not a marker of internal phase. The silver lining of this confusion between internal and external time is that our results suggest that chronotype does not seem to affect the identification or performance of the biomarker set. \\

Although chronotype is not relevant to predict skin phase in our cohort, we cannot extrapolate and apply the same rule for other tissues, or even in skin from extreme chronotypes. Archer et al. previously showed in \cite{Archer2014} that when sleep was scheduled out of circadian phase, the blood transcriptome was affected resulting in lower amplitude circadian, or even arrhythmic transcripts. This means that if a biomarker set includes such genes, it may fail to predict circadian phase in individuals that sleep out of their phase such as shift workers, the elderly and\slash or sick patients. Whether our proposed biomarker set is able to properly predict internal time in patients or cohorts with skin diseases, with known low circadian amplitudes, in extreme chronotypes and\slash or in the presence of internal circadian desynchronization remains to be elucidated. Additional unanswered questions, although out of the scope of this paper, are the elucidation of methods that might help to accurately and unobtrusively assess not just phase, but also amplitude, robustness and circadian disruption in order to understand the role of circadian rhythms in physical and mental health and disease. But unfortunately theconcepts of circadian amplitude or robustness are not well defined \cite{Dijk2020}, and a simple gold standard measure for circadian amplitude and robustness has not yet been agreed upon.\\

Any clinical use of circadian biomarkers needs fast, cost-effective and noninvasive methods such as hair follicle- or oral mucosa-sampling techniques, standardization across platforms (qPCR, NanoString, a limitation in this study) and ideally samples taken just once. Moreover, any algorithm assessing circadian phase should be validated against gold-standard markers of internal phase. For example, the phase of melatonin rhythm is considered the gold-standard marker for the phase of the SCN \cite{Pandi2007, Laing2017}. In order to unlock the future of circadian medicine and chronotherapy, larger studies should compare their findings with other techniques such as DLMO, actigraphy, etc. and confirm their results across a range of disease states and pathologies. Solving questions like these will undoubtedly help in defining good sources of biomarkers for application in circadian medicine. \\ %Any algorithm assessing this should be validated against gold-standard markers of internal phase. For example, the phase of melatonin rhythm is considered the gold-standard marker for the suprachiasmatic nucleus' phase. 

%\begin{itemize}
	%\item Epidermal clocks might entrain more efficiently because of their direct contact with experimental cues? Could it be that the higher amplitude rhythms are because of their contact with external cues?
	%\item Is skin clock phase a good predictor of clock phase in other tissues? What about monocytes in the BodyTime study?
	%\item Keep in mind that our sampling time is 4\,h and this might not be enough to capture subtle chronotype differences $\rightarrow$ reason why internal time versus wall time analysis give similar results?
	%\item \textcolor{red}{Dermis has a lot of fibroblasts $\rightarrow$ compare to Steve Brown's fibroblast paper}
	%\item \textcolor{red}{Talk about hair growth (anagen, telogen) in the context of epidermis rhythms}
	%\item \textcolor{red}{Benitah says that dermal compartment is important for ageing}
	%\item Are time-telling genes disrupted in skin-disease?	
	%\item vP results speaking for population sampling from skin (i.e., taking a skin biopsy just once) providing a good estimation of circadian time, regardless of wall time or internal time. \textit{Although this also has to do with our chronotype range $\sim$ sampling frequency...}
	%\item common circadian variation is strongest in our cohort so we expect these insights to translate across population.
	%\item not obvious/known that CCGs' magnitude varies quite a lot between subjects.
	%\item what our results mean for population rhythms vs individual rhythms
	%\item we hope to have captured a normal-enouqgh range of chronotypes
	%\item Is this similarity internal-external time circadian transcriptome something particular from our subjects? Or can we apply this similarity in time to any subject in general and to any population? Think.	
	%\item Wu-Hog combination to have higher 'power' and order time samples (CYCLOPS): they do Zeitzeiger vs external time and maybe internal makes more sense?
	%\item AMPLITUDE OF CORE CLOCK NETWORK DIFFERS BETWEEN LAYERS!
	%\item pHASE SPECIFICITY EVEN IN 2 SO CLOSELY LOCATED LAYERS!
	%\item A fundamental challenge in the analysis of complex high-throughput datasets is to quantify and interpret the contribution of different sources of variation. Many compelling questions concern the relationship between these sources of variation. How does the population or the different skin layers affect the genetic regulation of gene expression? Is technical variability of the microarray data low enough to study regulatory genetics and circadian biology? What are the major drivers of this technical variability?  \textcolor{red}{Rephrase}
	%\item comment on how good biomarkers should have correct phase, large amplitudes and same mean expression. So it should be genes with high circadian variation butby little mean variation across subjects (time series suppfig5A and check in vP). 
%\end{itemize}




